### PR TITLE
[DOCS] Document several missing thread pools

### LINKED ISSUE #48524: [DOCS] omission of core thread pools "management,  flush, force_merge, fetch_shard_started and fetch_shard_store"
**Describe the feature**: This is not feature request.  Just a document issue.

**Elasticsearch version** (`bin/elasticsearch --version`): latest

**Plugins installed**: []

**JVM version** (`java -version`): n/a

**OS version** (`uname -a` if on a Unix-like system): n/a

**Description of the problem including expected versus actual behavior**:

These core thread pools are undocumented as of now.

- management
- flush
- force_merge
- fetch_shard_started
- fetch_shard_store

please check here.
https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-threadpool.html

(ref)
https://github.com/elastic/elasticsearch/blob/d579f892dbd8e70acc27ecb95b9035a97352d8d8/server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java#L65

### PR DESCRIPTION
**Pull request by jrodewig**
Adds documentation for the following thread pools:

- `fetch_shard_started`
- `fetch_shard_store`
- `flush`
- `force_merge`
- `management`

Closes #48524

### CODE REVIEW

File: docs/reference/modules/threadpool.asciidoc [hunk 1]
```diff
@@ -52,6 +52,30 @@ There are several thread pools, but the important ones include:
     Mainly for java client executing of action when listener threaded is set to true.
     Thread pool type is `scaling` with a default max of `min(10, (# of available processors)/2)`.
 
+`fetch_shard_started`::
+    For listing shard states.
+    Thread pool type is `scaling` with keep-alive of `5m` and a default maximum
+    size of `2 * # of available processors`.
+
+`fetch_shard_store`::
+    For storing shard states.
```
Comment (jaymode) [inline docs/reference/modules/threadpool.asciidoc]:
```suggestion
    For listing shard stores.
```

File: docs/reference/modules/threadpool.asciidoc [hunk 2]
```diff
@@ -52,6 +52,30 @@ There are several thread pools, but the important ones include:
...
+    Thread pool type is `scaling` with keep-alive of `5m` and a default maximum
+    size of `2 * # of available processors`.
+
+`flush`::
+    For <<indices-flush,flush>> operations.
```
Comment (jaymode) [inline docs/reference/modules/threadpool.asciidoc]:
```suggestion
    For <<indices-flush,flush>>, <<indices-synced-flush-api,synced flush>>, and <<index-modules-translog, translog>> `fsync` operations.
```

File: docs/reference/modules/threadpool.asciidoc [hunk 3]
```diff
@@ -52,6 +52,30 @@ There are several thread pools, but the important ones include:
...
+    Thread pool type is `scaling` with a keep-alive of `5m` and a default
+    maximum size of `min(5, (# of available processors)/2)`.
+
+`force_merge`::
+    For <<indices-forcemerge,force merge>> operations.
+    Thread pool type is `scaling`.
```
Comment (jaymode) [inline docs/reference/modules/threadpool.asciidoc]:
```suggestion
    Thread pool type is `fixed` with a size of 1 and an unbounded queue size.
```

**Review by jaymode — APPROVED** [review]

**Comment by DaveCTurner** [comment]
@jrodewig it looks like this wasn't backported to 7.5, although it's there in the 7.4 and 7.6+ branches.

**Comment by jrodewig** [comment]
Thanks for pointing this out @DaveCTurner. That was an unintentional oversight on my part.

Backported to 7.5 with a813a27e591.

**Pull request merged**